addons/binding/org.openhab.binding.km200/src/main/java/org/openhab/binding/km200/handler/KM200ThingHandler.java

```diff
@@ -0,0 +1,451 @@
+                newChannel = createChannel(new ChannelTypeUID(thing.getUID().getAsString() + ":" + posName),
+                        new ChannelUID(thing.getUID(), posName), service + "/" + posName, "Number", currentPathName,
+                        "Positive switch of the cycle, like 'Day' 'On'", posName, true, true, state, "minutes");
+                subChannels.add(newChannel);
+
+                String negName = thing.getProperties().get(SWITCH_PROGRAM_NEGATIVE);
+                newChannel = createChannel(new ChannelTypeUID(thing.getUID().getAsString() + ":" + negName),
+                        new ChannelUID(thing.getUID(), negName), service + "/" + negName, "Number", currentPathName,
+                        "Negative switch of the cycle, like 'Night' 'Off'", negName, true, true, state, "minutes");
+                subChannels.add(newChannel);
+
+            }
+            ThingBuilder thingBuilder = editThing();
+            List<Channel> actChannels = thing.getChannels();
+            for (Channel channel : actChannels) {
+                thingBuilder.withoutChannel(channel.getUID());
+            }
+            thingBuilder.withChannels(subChannels);
+            updateThing(thingBuilder.build());
+            isInited = true;
+            updateStatus(ThingStatus.ONLINE);
+        }
+    }
+
+    @Override
+    public void handleRemoval() {
+        logger.debug("Handle removed");
+        this.updateStatus(ThingStatus.REMOVED);
+    }
+
+    @Override
+    public void bridgeStatusChanged(ThingStatusInfo bridgeStatusInfo) {
+        if (bridgeStatusInfo.getStatus().equals(ThingStatus.ONLINE)) {
+            initialize();
+        } else if (bridgeStatusInfo.getStatus() == ThingStatus.OFFLINE) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE);
+        } else if (bridgeStatusInfo.getStatus() == ThingStatus.UNKNOWN) {
+            updateStatus(ThingStatus.UNKNOWN);
+        }
+    }
+
+    @Override
+    public void dispose() {
+        channelTypeProvider.removeChannelTypesForThing(getThing().getUID());
+    }
+
+    /**
+     * Checks whether a channel is linked to an item
+     *
+     * @param channel
+     */
+    public Boolean checkLinked(Channel channel) {
+        return isLinked(channel.getUID().getId());
+    }
+
+    /**
+     * Search for services and add them to a list
+     *
+     * @param serObj
+     * @param thing
+     * @param subChannels
+     * @param subNameAddon
+     */
+    private void addChannels(KM200CommObject serObj, Thing thing, List<Channel> subChannels, String subNameAddon) {
+        String service = serObj.getFullServiceName();
+        Set<String> subKeys = serObj.serviceTreeMap.keySet();
+        List<String> asProperties = null;
+        /* Some defines for dummy values, we will ignore such services */
+        final BigDecimal maxInt16AsFloat = new BigDecimal(+3276.8).setScale(6, RoundingMode.HALF_UP);
```

Comment:
What about making them constants and longs :-)

Reply:
They are float values. But I would like to let it like it is. It's only used in this one function.